components/brave_ads/browser/ads_service_callback.h

```diff
@@ -19,13 +20,14 @@ struct NewTabPageAdInfo;
 namespace brave_ads {
 
 using GetDiagnosticsCallback =
-    base::OnceCallback<void(const bool, const std::string&)>;
+    base::OnceCallback<void(absl::optional<base::Value::List> value)>;
```

Comment:
Nit: I'd probably just omit `value` here (and below).

Reply:
Fixed